source/security/auth.txt

```diff
@@ -0,0 +1,160 @@
+.. _scala-auth:
+
+=========================
+Authentication Mechanisms
+=========================
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: code example, verify, AWS, Kerberos
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+The driver supports all MongoDB authentication mechanisms,
+including those available only in the MongoDB Enterprise Edition.
+
+MongoCredential
+---------------
+
+An authentication credential is represented as an instance of the
+``MongoCredential`` class. The ``MongoCredential`` class includes
+static factory methods for each of the supported authentication
+mechanisms.
+
+Default Authentication Mechanism
+--------------------------------
+
+In MongoDB 3.0, MongoDB changed the default authentication mechanism
+from ``MONGODB-CR`` to ``SCRAM-SHA-1``. In MongoDB 4.0, support for
+the deprecated ``MONGODB-CR`` mechanism was removed and ``SCRAM-SHA-256`` support was
+added.
+
+To create a credential that authenticates by using the default
+authentication mechanism, regardless of server version, create a
+credential by using the ``createCredential()`` static factory method:
+
+.. literalinclude:: /includes/security/auth.scala
+   :start-after: start-default
+   :end-before: end-default
+   :language: scala
+   :copyable:
+   :dedent:
+
+Or, you can use a connection string without explicitly specifying the
+authentication mechanism:
+
+.. literalinclude:: /includes/security/auth.scala
+   :start-after: start-default-connection-string
+   :end-before: end-default-connection-string
+   :language: scala
+   :copyable:
+   :dedent:
+
+For challenge and response mechanisms, using the default
+authentication mechanism is the recommended approach, as it makes
+upgrading from MongoDB 2.6 to MongoDB 3.0 more simple, even after
+upgrading the authentication schema. For MongoDB 4.0 users, using the
+default authentication mechanism is also recommended as the mechanisms are
+checked and the correct hashing algorithm is used.
```

Comment:
[s] Can potentially edit this to be more concise per [the style guide](https://www.mongodb.com/docs/meta/style-guide/terminology/concise-terms/) and use active voice. Something like: 
```suggestion
For challenge and response mechanisms, we recommend using the default
authentication mechanism.  This approach simplifies upgrading from MongoDB 
2.6 to MongoDB 3.0, even after upgrading the authentication schema. For 
MongoDB 4.0 users, the default authentication mechanism is also 
recommended because it checks the mechanisms and uses the correct hashing algorithm.
```